br/cmd/br/cmd.go

```diff
@@ -107,6 +110,23 @@ func AddFlags(cmd *cobra.Command) {
 	_ = cmd.PersistentFlags().MarkHidden(FlagRedactLog)
 }
 
+func calculateMemoryLimit(memleft uint64) uint64 {
+	const halfGB = uint64(512 * 1024 * 1024)
+	const fourGB = 8 * halfGB
+	// memreserved = f(memleft) = 512MB * memleft / (memleft + 4GB)
+	//  * f(0) = 0
```

Comment:
if the mem unused is 0, do we even bother to start the BR? I guess a correctly configured machine should not use up all its mem and if they do they are facing a bigger problem and we probably don't want to start a BR to make things more complicated. Let me know your thoughts!

Reply:
I think maybe it is already OOM. Maybe we can directly limit the minimum remaining memory.

Reply:
yeah, I feel like we can have some minimum memory not starting from 0, if below that we just throw error and exit.

Reply:
I add the minimum value (`256MiB`) of memory limit. It's hard to determine the minimum memory br usage. In my test, br uses only about 200 MB~300 MB memory to backup data generated by `tiup bench tpcc prepare --warehouses 3`.

Reply:
thanks for digging into that, yeah having an estimation of memory usage very useful.